### DISCUSSION TITLE
Roadmap upgrade react-virtual v2 to v3

### DISCUSSION THREAD

**Discussion by josecao** [KevinVandy/material-react-table#122] [General]
Hi admin, what do you plan to upgrade v2 to v3 for this lib ?

**Comment by KevinVandy** [comment]
I have tried to upgrade to the v3 beta of react-virtual, but it had too many bugs so I had to revert it. 

`@tanstack/react-virtual` v3 isn't even out as stable yet, so won't consider upgrading until they release.

**Reply by djdmbrwsk** [reply]
Would you consider publishing a beta of `material-react-table`? 😄 I'm getting some peer dependency issues with React 18 while trying to play with Next 13 😕

I updated `react-virtual` (and `virtual-core`) and after some very brief `npm link` testing with one of my projects it seems to be working decently. Branch here https://github.com/djdmbrwsk/material-react-table/tree/wip/update-react-virtual

**Reply by KevinVandy** [reply]
I 100% would love to upgrade to the latest beta of react-virtual if it works. I was still running into lots of bugs last time I tried it.

For MRT 2.0, I'm considering reserving the virtualization feature only when the table renders with divs instead of table tags. Still considering a lot of things

**Reply by KevinVandy** [reply]
MRT v1.4.0 is now on `@tanstack/react-virtual` v3

**Comment by andresgutgon** [comment]
Hi @KevinVandy first of all thanks for this project. We're using similar stack `@tanstask/react-table` and `@tanstack/react-virtual` and this project it's a great reference for us so Thanks!

Regarding this discussion. What were the issues you found that made you revert  to old version?

I'm trying new version and looks it's working

**Reply by KevinVandy** [reply]
If you're able to make the code shareable for it working, I'd be really interested. The major bugs with the approach I took here comes were that the table would start vibrating up and down randomly. And scrolling the rows while the table was halfway out of the viewport would result in the entire webpage shaking up and down. Kind of strange.

If you are using a more traditional `<div>` datagrid though, you may not have as many problems.

**Reply by andresgutgon** [reply]
Yes, we're building a data grid virtualized horizontal and vertical with divs no tables. I did nothing special just follow docs

**Reply by KevinVandy** [reply]
MRT v1.4.0 is now on `@tanstack/react-virtual` v3

**Comment by KevinVandy** [comment]
MRT v1.4.0 is now on `@tanstack/react-virtual` v3
